internal/manifest/report.go

```diff
@@ -66,17 +72,55 @@ func (r *Report) Add(slice *setup.Slice, fsEntry *fsutil.Entry) error {
 		r.Entries[relPath] = entry
 	} else {
 		r.Entries[relPath] = ReportEntry{
-			Path:   relPath,
-			Mode:   fsEntry.Mode,
-			SHA256: fsEntry.SHA256,
-			Size:   fsEntry.Size,
-			Slices: map[*setup.Slice]bool{slice: true},
-			Link:   fsEntry.Link,
+			Path:       relPath,
+			Mode:       fsEntry.Mode,
+			SHA256:     fsEntry.SHA256,
+			Size:       fsEntry.Size,
+			Slices:     map[*setup.Slice]bool{slice: true},
+			Link:       fsEntry.Link,
+			HardLinkId: hardLinkId,
 		}
 	}
 	return nil
 }
 
+func (r *Report) handleHardlinkReport(fsEntry *fsutil.Entry) (*fsutil.Entry, int) {
+	// Handle the hard link group
+	hardLinkId := NON_HARDLINK
+	sha256 := fsEntry.SHA256
+	size := fsEntry.Size
+	link := fsEntry.Link
+	if r.entryIsHardLink(fsEntry.Link) {
+		relLinkPath, _ := r.sanitizeAbsPath(fsEntry.Link, false)
+		// With this, a hardlink is found
+		if entry, ok := r.Entries[relLinkPath]; ok {
+			if entry.HardLinkId == NON_HARDLINK {
+				r.currHardLinkId++
+				entry.HardLinkId = r.currHardLinkId
+				r.Entries[relLinkPath] = entry
+			}
+			hardLinkId = entry.HardLinkId
+			if fsEntry.Mode.IsRegular() {
+				// The hard link links to a regular file
+				sha256 = entry.SHA256
+				size = entry.Size
+				link = ""
```

Comment:
```suggestion
				fsEntry.Link = ""
```

Reply:
^^ when we change the `fsEntry` from within.

Reply:
Also, why did we need to update these two?

```go
				sha256 = entry.SHA256
				size = entry.Size
```

Reply:
Marked as resolved. Explained offline.